src/bll/weblinklistener/httphostrequest.ts

```diff
@@ -0,0 +1,98 @@
+    }
+
+    public async processRequest(bytes: Uint8Array): Promise<{succeed: boolean, httpVersion: string}> {
+        const input: string[] = bytes.toString().trim().split("\n");
+        const firstLine: string = input[0];
+        Logger.logDebug(`HttpHostRequest#processRequest: firstLine is ${firstLine}`);
+        const tokens: string[] = firstLine.trim().split(/[\t\r\n\f ]+/);
+        if (tokens.length < 3) {
+            Logger.logWarning(`HttpHostRequest#processRequest: not enough tokens: ${tokens.join(";")}`);
+            return {succeed: false, httpVersion: undefined};
+        }
+
+        const uri = tokens[1];
+        const httpVersion = tokens[2];
+        let path = uri;
+        const params = {};
+
+        const endOfPathIndex: number = uri.indexOf("?");
+        if (endOfPathIndex >= 0) {
+            path = uri.substring(0, endOfPathIndex);
+            const request = uri.substring(endOfPathIndex + 1);
+            this.parseGetRequestParams(request, params);
+        }
+
+        const succeed = await this.fireRequestAccepted(path, params);
+        return {succeed: succeed, httpVersion: httpVersion};
+    }
+
+    private parseGetRequestParams(encodedURIComponent: string, pars: {}): void {
+        if (!encodedURIComponent) {
+            return;
+        }
+
+        const req = decodeURIComponent(encodedURIComponent);
+        const pairs: string[] = req.split("&");
+        pairs.forEach((pair) => {
+            const s: number = pair.indexOf("=");
+            if (s > 0) {
+                const key: string = pair.substring(0, s);
+                const value: string = pair.substring(s + 1);
+                if (key && value) {
+                    pars[key] = value;
+                }
+            }
+        });
+    }
+
+    private async fireRequestAccepted(resourcePath: string, pars: {}): Promise<boolean> {
+        const data: HttpRequestData = new HttpRequestData(resourcePath, pars);
+        const workspaceFolders: any[] = this.workspaceProxy.getWorkspaceFolders();
+        if (!data.isSupportedRequest() || !workspaceFolders || workspaceFolders.length === 0) {
+            Logger.logInfo("HttpHostRequest#fireRequestAccepted: either request not supported either wf.len = 0");
+            return false;
+        }
+
+        for (let i = 0; i < workspaceFolders.length; i++) {
+            const filePath = path.join(workspaceFolders[i].uri.path, data.getFile());
+            if (await this.tryShowTextDocument(filePath)) {
+                Logger.logDebug(`HttpHostRequest#fireRequestAccepted: ${filePath} is opened!`);
+                return true;
+            }
+            Logger.logDebug(`HttpHostRequest#fireRequestAccepted: attempt of opening ${filePath} has failed`);
+        }
+        return false;
+    }
+
+    private async tryShowTextDocument(filePath: string) {
+        const uriPath = this.uriProxy.file(filePath);
+        if (await this.fsProxy.existsAsync(uriPath.fsPath)) {
```

Comment:
Also maybe valuable to report that file does not exists

Reply:
Do you mean to add logging of the information?
Currently there is Logger.logDebug(`HttpHostRequest#fireRequestAccepted: attempt of opening ${filePath} has failed`); that indicates that file doesn't exist.

Reply:
@rugpanov, ok